<?xml version="1.0" ?>
<robot name="cart">
  <link name="world"/>
  <link name="slideBar">
    <visual>
      <geometry>
        <box size="30 0.05 0.05"/>
      </geometry>
      <origin xyz="0 0 0"/>
      <material name="green">
        <color rgba="0 0.8 .8 1"/>
      </material>
    </visual>
    <inertial>
      <mass value="100"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <link name="cart">
    <visual>
      <geometry>
        <box size="0.5 0.5 0.2"/>
      </geometry>
      <origin xyz="0 0 0"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="world_to_base" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 1"/>
    <parent link="world"/>
    <child link="slideBar"/>
  </joint>
  <joint name="joint_1" type="prismatic">
    <axis xyz="1 0 0"/>
    <origin xyz="0.0 0.0 0.0"/>
    <parent link="slideBar"/>
    <child link="cart"/>
    <limit effort="1000.0" lower="-15" upper="15" velocity="30"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>

  <ros2_control name="linrob_hw" type="system">
    <hardware>
        <plugin>linrob/Resource</plugin>
        <!-- Datalayer Connection -->
        <param name="ip">10.0.2.2</param>
        <param name="user">boschrexroth</param>
        <param name="password">boschrexroth</param>
        <param name="ssl_port">5443</param>

        <!-- Datalayer Nodes addresses -->
        <param name="new_position">plc/app/Application/sym/GVL/alrAxisTargetPositionsExt</param>
        <param name="new_position_timestamp">plc/app/Application/sym/GVL/alrAxisTargetPositionTimestampExt</param>
        <param name="next_pos_index">plc/app/Application/sym/GVL/nAxisTargetPosExtIdx</param>
        <param name="execute_movements">plc/app/Application/sym/GVL/xAxisTargetPosExtExecute</param>
        <param name="position">plc/app/Application/sym/GVL/lrAxisPositionX</param>
        <param name="velocity">plc/app/Application/sym/GVL/lrAxisVelocityX</param>
        <param name="status">plc/app/Application/sym/GVL/anAxisStatus</param>
        <param name="read_mode">plc/app/Application/sym/GVL/sOperationMode</param>
        <param name="set_mode">plc/app/Application/sym/GVL/nOperationModeRequest</param>

        <!-- Virtual command variables for axis control services -->
        <param name="virtual_reset">plc/app/Application/sym/GVL/xResetVirt</param>
        <param name="virtual_reference">plc/app/Application/sym/GVL/xReferenceVirt</param>
        <param name="virtual_stop">plc/app/Application/sym/GVL/xStopVirt</param>

        <param name="error_code">plc/app/Application/sym/GVL/nLatestErrorCode</param>

        <param name="initial_position_index">1</param>
        <param name="max_position_indices">1000</param>
        <param name="execute_movements_on_next_index">3</param>

        <param name="update_frequency_hz">10</param>

        <param name="log_level">INFO</param>


    </hardware>
    <joint name="joint_1">
      <command_interface name="position">
        <param name="min">-15</param>
        <param name="max">15</param>
      </command_interface>
      <command_interface name="virtual_reset"/>
      <command_interface name="virtual_reference"/>
      <command_interface name="virtual_stop"/>
      <state_interface name="position">
        <param name="initial_value">1.0</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="error_code"/>
    </joint>
  </ros2_control>
</robot>
